mlir/lib/Dialect/Bufferization/IR/BufferizationOps.cpp

```diff
@@ -505,6 +505,11 @@ struct SimplifyClones : public OpRewritePattern<CloneOp> {
     // of the source.
     for (Operation *pos = cloneOp->getNextNode(); pos != redundantDealloc;
          pos = pos->getNextNode()) {
+      // If the next node is nullptr, it indicates there is no redundantDealloc
+      // after cloneOp, means that there is a redundantDealloc preseding
```

Comment:
`preceding`

Reply:
I think it could also mean that there is no dealloc at all. Can we phrase this along the lines of `Bail if we run out of operations while looking for a deallocation op.`?